test/query_test.rb

```diff
@@ -32,9 +39,21 @@ def test_find
     assert_equal [], User.find(name: "Noah").to_a
   end
 
+  def test_find!
+    user = User.create(name: "Bryan")
+    assert_equal user, User.find!(user.id)
+    assert_raises(Minidoc::DocumentNotFoundError) { User.find!(BSON::ObjectId.new) }
```

Comment:
:)

after writing I confirmed the assertion won't succeed (because it will return a mongo cursor that has no matching documents, which is truthy). Im not sure how important that consistency is -- but might be worth documenting the expected behavior via a test either way